main/src/ca/uwaterloo/flix/language/phase/jvm/JvmBackend.scala

```diff
@@ -145,10 +145,35 @@ object JvmBackend extends Phase[Root, CompilationResult] {
       val lazyClasses = GenLazyClasses.gen(types)
 
       //
-      // Generate Unit class.
+      // Generate the Unit class.
       //
       val unitClass = GenUnitClass.gen()
 
+      //
+      // Generate the FlixError class.
+      //
+      val flixErrorClass = GenFlixErrorClass.gen()
+
+      //
+      // Generate the ReifiedSourceLocation class.
+      //
+      val rslClass = GenReifiedSourceLocationClass.gen()
+
+      //
+      // Generate the HoleError class.
+      //
+      val holeErrorClass = GenHoleErrorClass.gen()
+
+      //
+      // Generate the MatchError class.
+      //
+      val matchErrorClass = GenSimpleLocationErrorClass.gen(JvmName.MatchError, "Non-exhaustive match at ")
```

Comment:
I am not sure I understand this design pattern.

Reply:
GenSimpleLocationErrorClass.gen makes you a subclass of Flix with the given name, where the string given is prefixed on the message if the error is thrown. This could also be a single class that has a string constructor argument (like HoleError i suppose) but i initially just wanted to recreate the existing classes.